Account/FirefoxAccount.swift

```diff
@@ -190,13 +195,79 @@ open class FirefoxAccount {
             return "Not in a Token State: \(state?.label.rawValue ?? "Empty State")"
         }
     }
-
+    
+    public class FxAProfile {
+        open var avatarUrl: URL?
+        open var avatarImageData: Data?
+        open var displayName: String?
+        open let email: String
+        
+        init(email: String, displayName: String?, avatar: String?) {
+            self.email = email
+            self.displayName = displayName
+            self.avatarImageData = nil
+
+            if let avatarStr = avatar {
+                self.avatarUrl = URL(string: avatarStr)
+            } else {
+                self.avatarUrl = nil
+            }            
+        }
+        
+        func downloadAvatarImage() {
+            guard let avatarUrl = self.avatarUrl else {
+                return
+            }
+            
+            DispatchQueue.global().async {
+                self.avatarImageData = try? Data(contentsOf: avatarUrl)
+                NotificationCenter.default.post(name: NotificationFirefoxAccountProfileChanged, object: self)
+            }
+        }
+    }
+    
+    // Fetch current user's FxA profile. It contains the most updated email, displayName and avatar.
+    open func updateProfile() {
+        guard let session = stateCache.value as? TokenState else {
+            return
+        }
+        
+        let client = FxAClient10(authEndpoint: self.configuration.authEndpointURL, oauthEndpoint: self.configuration.oauthEndpointURL, profileEndpoint: self.configuration.profileEndpointURL)
+        return client.getProfile(withSessionToken: session.sessionToken as NSData) >>== { result in
+            
+            if let fxaProfile = self.fxaProfile {
+                // In the case there is an existing FxA profile, only update the changed properties.
+                if fxaProfile.email != result.email {
+                    self.fxaProfile = FxAProfile(email: result.email, displayName: result.displayName, avatar: result.avatar)
+                    self.fxaProfile?.downloadAvatarImage()
+                }
+                
+                fxaProfile.displayName = result.displayName
```

Comment:
You don't need to do all of this work if you just replaced the profile object, right?